dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorityRestRepositoryIT.java

```diff
@@ -210,6 +211,165 @@ public void retrieveSolrValueTest() throws Exception {
+                 .andExpect(jsonPath("$.page.number", is(0)));
+
+        // second page
+        getClient(tokenAdmin).perform(get("/api/integration/authorities/srsc/entries/search/byParent")
+                .param("id", "SCB14")
+                .param("page", "1")
+                .param("size", "2"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$._embedded.authorityEntries", Matchers.containsInAnyOrder(
+                    AuthorityEntryMatcher.matchAuthority("SCB1409", "Other mathematics")
+                    )))
+                .andExpect(jsonPath("$.page.totalElements", is(3)))
+                .andExpect(jsonPath("$.page.totalPages", is(2)))
+                .andExpect(jsonPath("$.page.number", is(1)));
+    }
+
+    @Test
+    public void srscSearchByParentSecondLevel_Applied_mathematics_Test() throws Exception {
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+        getClient(tokenAdmin).perform(get("/api/integration/authorities/srsc/entries/search/byParent")
+                             .param("id", "SCB1402"))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$._embedded.authorityEntries", Matchers.containsInAnyOrder(
+                                     AuthorityEntryMatcher.matchAuthority("VR140202", "Numerical analysis"),
+                                     AuthorityEntryMatcher.matchAuthority("VR140203", "Mathematical statistics"),
+                                     AuthorityEntryMatcher.matchAuthority("VR140204", "Optimization, systems theory"),
+                                     AuthorityEntryMatcher.matchAuthority("VR140205", "Theoretical computer science")
+                                     )))
+                             .andExpect(jsonPath("$.page.totalElements", Matchers.is(4)));
+    }
+
+    @Test
+    public void srscSearchByParentEmptyTest() throws Exception {
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+        getClient(tokenAdmin).perform(get("/api/integration/authorities/srsc/entries/search/byParent")
+                             .param("id", "VR140202"))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$.page.totalElements", Matchers.is(0)));
+    }
+
+    @Test
+    public void srscSearchByParentWrongIdTest() throws Exception {
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+        getClient(tokenAdmin).perform(get("/api/integration/authorities/srsc/entries/search/byParent")
+                             .param("id", "WRONG_ID"))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$.page.totalElements", Matchers.is(0)));
+    }
+
+    @Test
+    public void srscSearchTest() throws Exception {
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
```

Comment:
It looks like all these tests currently only run as an Admin user. We'd need tests that also run as a normal authenticated user (which should have permissions too), and also tests to verify that an anonymous user has no access.